File: Object_Oriented_Programming/polymorphism.py
#polymorphism
#more than one form
#polymorphism includes:
#method overloading
#method overriding
#operator overloading

#method overloading
#same method with different number of arguments


class Math:
    def add(self):
        print("No arguments")

    def add(self,num1):
        print("one arguments")

    def add(self,num1,num2):
        print("two arguments")

m=Math()
m.add(10,20)

#In method overloading only the recently implemented methods only works

File: Html_sections/htmlsections/luproject/luminar-project/Javascript/Object_Oriented_programs/polymorphism.js
//method overloading
//method overloading(same method name and different number of arguements)

class Calculate{
    add(){
        console.log("No arguement method");
    }
    add(num){
        console.log("One arguement method");
    }
    add(num1,num2){
        console.log("Two arguement method");
    }
}

var obj=new Calculate()
obj.add(10);
obj.add(20,10);

